feat(content): introduce schema types

damienwebdev opened the pull request:
## PR Checklist

- [x] Commit message follows our [contributing guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit)
- [x] Tests added/updated (for bug fixes/features)
- [x] Documentation added/updated (for bug fixes/features)

## PR Type
<!-- Select the relevant type(s) -->

- [ ] Bug fix
- [x] Feature
- [ ] Style update
- [ ] Refactor
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Docs
- [ ] Performance
- [ ] Other (please describe)

## Current behavior
Currently, we have no way to render content from ecommerce to Daffodil storefronts beyond simple HTML.

## New behavior
This defines an AST so that we can bring content from these platforms in a structured format without requiring custom elements.

## Breaking change?

- [ ] Yes
- [x] No

<!-- If yes, describe impact and migration path -->

## Additional context
<!-- Any other relevant information -->

damienwebdev committed:
feat(content): introduce schema types

damienwebdev committed:
Remove component registration section from schema.md

Removed the section on registering components, including code examples and explanations.

griest024 commented on libs/content/guides/schema.md:
````diff
@@ -0,0 +1,227 @@
+```typescript
+const textSchema: DaffTextSchema = {
+  type: 'textSchema',
+  text: 'This is a paragraph of text.'
````
How is it actually rendered in dom? As a p?

damienwebdev replied:
There's a small discrepancy on this point between the "editable-renderer" (which admins use to modify content) and the "renderer".

- `editable-renderer` renders `<span contenteditable>This is a paragraph of text.</span>`
- `schema-renderer` renders `This is a paragraph of text.` inside the renderer root

griest024 replied:
I think its misleading to word it like that then, reading this I would assume that it gets rendered as `<p>`.  Just change the content of the text to not reference a paragraph

griest024 commented on libs/content/guides/schema.md:
```diff
@@ -0,0 +1,227 @@
+
+`DaffContentElementSchema` define HTML elements with optional attributes, styles, and children.
+
+### Example: Basic Element
```
I think all of these sections should show the interface, if any of them do

damienwebdev replied:
What do you mean? I want to leave that to the API docs so that I don't have to maintain this content.

griest024 replied:
Seems good, what is your motivation for showing the text schema interface?

damienwebdev replied:
I removed it.

griest024 commented on libs/content/guides/schema.md:
```diff
@@ -0,0 +1,227 @@
+const productCardSchema: DaffContentComponentSchema = {
+  type: 'componentSchema',
+  name: 'ProductCard',
+  inputs: {
```
Can I bind dynamic values?

damienwebdev replied:
No. Schema is purely static. You could re-compute schema dynamically if you'd like.

griest024 replied:
Its probably worth adding a note here to that effect.

damienwebdev replied:
Added a note.

griest024 commented on libs/content/guides/schema.md:
```diff
@@ -0,0 +1,227 @@
+    title: 'Featured Product',
+    variant: 'elevated'
+  },
+  children: [
```
Does this support all the content projection features available when authoring traditional angular templates?

damienwebdev replied:
It doesn't currently. That's a feature target that I want to work towards. Currently everything is transcluded into the main `ng-content` of the `componentSchema`.

That said, I don't think the schema would be the one defining which slot your element is targeting. Depending on the attributes (just like with regular Angular templates) certain slots would be targeted.

griest024 replied:
I was mainly thinking of `exportAs`, but yeah, we'll want this full feature support eventually

damienwebdev committed:
Update text in textSchema example

damienwebdev committed:
Remove DaffTextSchema interface from schema.md

Removed interface definition for DaffTextSchema from the text schema documentation.

damienwebdev committed:
Update schema description for clarity on static nature

Clarified that the content schema is static and does not support functions or callbacks.
